planner/core/plan_cost_ver2.go

```diff
@@ -617,40 +665,30 @@ func getTaskScanFactor(p PhysicalPlan, taskType property.TaskType) float64 {
 			desc = tableScan.Desc
 			tbl = tableScan.Table
 		}
+		if tbl != nil && tbl.TempTableType != model.TempTableNone {
+			return defaultVer2Factors.TiDBTempScan // temp table
+		}
 		if desc {
-			return p.SCtx().GetSessionVars().GetDescScanFactor(tbl)
+			return defaultVer2Factors.TiKVDescScan
 		}
-		return p.SCtx().GetSessionVars().GetScanFactor(tbl)
+		return defaultVer2Factors.TiKVScan
 	}
 }
 
-func getTaskNetFactor(p PhysicalPlan, _ property.TaskType) float64 {
-	// TODO: introduce a dedicated net factor for TiFlash
-	return p.SCtx().GetSessionVars().GetNetworkFactor(getTableInfo(p))
-}
-
-func getTaskSeekFactor(p PhysicalPlan, _ property.TaskType) float64 {
-	return p.SCtx().GetSessionVars().GetSeekFactor(getTableInfo(p))
-}
-
-func getTableInfo(p PhysicalPlan) *model.TableInfo {
-	switch x := p.(type) {
-	case *PhysicalIndexReader:
-		return getTableInfo(x.indexPlan)
-	case *PhysicalTableReader:
-		return getTableInfo(x.tablePlan)
-	case *PhysicalIndexLookUpReader:
-		return getTableInfo(x.tablePlan)
-	case *PhysicalTableScan:
-		return x.Table
-	case *PhysicalIndexScan:
-		return x.Table
-	default:
-		if len(x.Children()) == 0 {
-			return nil
+func getTaskNetFactorVer2(p PhysicalPlan, _ property.TaskType) costVer2Factor {
+	if _, ok := p.(*PhysicalExchangeReceiver); ok { // TiFlash MPP
+		return defaultVer2Factors.TiFlashMPPNet
+	}
+	if tblReader, ok := p.(*PhysicalTableReader); ok {
+		if _, isMPP := tblReader.tablePlan.(*PhysicalExchangeSender); isMPP { // TiDB to TiFlash with mpp protocol
+			return defaultVer2Factors.TiDB2FlashNet
 		}
-		return getTableInfo(x.Children()[0])
 	}
+	return defaultVer2Factors.TiDB2KVNet
+}
+
+func getTaskSeekFactorVer2(p PhysicalPlan, _ property.TaskType) costVer2Factor {
+	return defaultVer2Factors.TiDBRequest
 }
```

Comment:
I think the logic changed here.
1. It doesn't recursively find `TableInfo` in the children now.
2. It doesn't return 0 for the temporary table now.